miasm2/arch/aarch64/sem.py

```diff
@@ -0,0 +1,745 @@
+
+
+@sbuild.parse
+def eor(arg1, arg2, arg3):
+    arg1 = arg2 ^ extend_arg(arg2, arg3)
+
+
+@sbuild.parse
+def eon(arg1, arg2, arg3):
+    arg1 = arg2 ^ (~extend_arg(arg2, arg3))
+
+
+@sbuild.parse
+def orr(arg1, arg2, arg3):
+    arg1 = arg2 | extend_arg(arg2, arg3)
+
+
+@sbuild.parse
+def orn(arg1, arg2, arg3):
+    arg1 = arg2 | (~extend_arg(arg2, arg3))
+
+
+@sbuild.parse
+def bic(arg1, arg2, arg3):
+    arg1 = arg2 & (~extend_arg(arg2, arg3))
+
+
+@sbuild.parse
+def mvn(arg1, arg2):
+    arg1 = (~extend_arg(arg1, arg2))
+
+
+def adds(ir, instr, arg1, arg2, arg3):
+    e = []
+    arg3 = extend_arg(arg2, arg3)
+    res = arg2 + arg3
+    e += update_flag_arith(res)
+    e += update_flag_add(arg2, arg3, res)
+    e.append(m2_expr.ExprAff(arg1, res))
+    return e, []
+
+
+def subs(ir, instr, arg1, arg2, arg3):
+    e = []
+    arg3 = extend_arg(arg2, arg3)
+    res = arg2 - arg3
+    e += update_flag_arith(res)
+    e += update_flag_sub(arg2, arg3, res)
+    e.append(m2_expr.ExprAff(arg1, res))
+    return e, []
+
+
+def cmp(ir, instr, arg1, arg2):
+    e = []
+    arg2 = extend_arg(arg1, arg2)
+    res = arg1 - arg2
+    e += update_flag_arith(res)
+    e += update_flag_sub(arg1, arg2, res)
+    return e, []
+
+
+def cmn(ir, instr, arg1, arg2):
+    e = []
+    arg2 = extend_arg(arg1, arg2)
+    res = arg1 + arg2
+    e += update_flag_arith(res)
+    e += update_flag_add(arg1, arg2, res)
+    return e, []
+
+
+def ands(ir, instr, arg1, arg2, arg3):
+    e = []
+    arg3 = extend_arg(arg2, arg3)
+    res = arg2 & arg3
+    e += update_flag_logic(res)
+    e.append(m2_expr.ExprAff(arg1, res))
+    return e, []
+
+
+@sbuild.parse
+def lsl(arg1, arg2, arg3):
+    arg1 = arg2 << (arg3 & m2_expr.ExprInt_from(arg3, arg3.size - 1))
+
+
+@sbuild.parse
+def lsr(arg1, arg2, arg3):
+    arg1 = arg2 >> (arg3 & m2_expr.ExprInt_from(arg3, arg3.size - 1))
+
+
+@sbuild.parse
+def asr(arg1, arg2, arg3):
+    arg1 = m2_expr.ExprOp(
+        'a>>', arg2, (arg3 & m2_expr.ExprInt_from(arg3, arg3.size - 1)))
+
+
+@sbuild.parse
+def mov(arg1, arg2):
+    arg1 = arg2
+
+
+def movk(ir, instr, arg1, arg2):
+    e = []
+    if isinstance(arg2, m2_expr.ExprOp):
+        assert(arg2.op == 'slice_at' and
+               isinstance(arg2.args[0], m2_expr.ExprInt) and
+               isinstance(arg2.args[1], m2_expr.ExprInt))
+        value, shift = int(arg2.args[0].arg), int(arg2.args[1].arg)
+        e.append(
+            m2_expr.ExprAff(arg1[shift:shift + 16], m2_expr.ExprInt16(value)))
+    else:
+        e.append(m2_expr.ExprAff(arg1[:16], m2_expr.ExprInt16(int(arg2.arg))))
+
+    return e, []
+
+
+@sbuild.parse
+def movz(arg1, arg2):
+    arg1 = arg2
+
+
+@sbuild.parse
+def movn(arg1, arg2):
+    arg1 = ~arg2
+
+
+@sbuild.parse
+def bl(arg1):
+    PC = arg1
+    ir.IRDst = arg1
+    LR = m2_expr.ExprInt64(instr.offset + instr.l)
+
+
+def csel(ir, instr, arg1, arg2, arg3, arg4):
+    e = []
+    cond_expr = cond2expr[arg4.name]
```

Comment:
Is this kind of expression unsupported in `@sbuild.parse` ?
If so, please open a feature request. These functions (with `csinc`, `csinv`, ...) may be more understandable as:
```Python
arg1 = cond_expr[arg4.name] if arg2 else arg3
```